src/Sylius/Bundle/CoreBundle/EventListener/Workflow/OrderCheckout/CreateOrderListener.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\EventListener\Workflow\OrderCheckout;
+
+use Sylius\Bundle\CoreBundle\StateMachine\StateMachineInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Order\OrderTransitions;
+use Symfony\Component\Workflow\Event\CompletedEvent;
+use Webmozart\Assert\Assert;
+
+final class CreateOrderListener
```

Comment:
I also don't have a better name. I'll leave this name, and we will see what other say.

Reply:
How about `CompleteOrderListener`?

Reply:
As I see below, there is `SaveCheckoutCompletionDateListener` listener, so `OrderCompletionListener` would be good I think

Reply:
The last suggestion, `CompleteOrder` could be misleading, because here we are completing a checkout and creating an order, not completing an order.
I have no better ideas than those above (Create or Place), or maybe, if we want to be precise, simply: `ApplyCreateTransitionOnOrder` or sth similar 🤔

Reply:
Since we're calling these listeners by what they're doing, maybe we should go a bit more verbose and simply name it `TransitionCartToOrderListener` or alike since that's what is actually happening here.

Reply:
> How about `CompleteOrderListener`?

If name will be derived from a transition, then `CompleteOrderCheckoutListener` will be appropriate, but we have 4 callbacks on `complete` transition and each does something else, so current naming scheme is probably for the best, even if it can confusing by reading from classes instead of winzou SM yaml config.

Reply:
Thank you for the name suggestions :) I think `ApplyCreateTransitionOnOrder` is consistent with other names.